### DISCUSSION TITLE
Help with optimizing config. Xeon E5-2673v3 2,4-2,7x2, ram 128Gb, 3x

### DISCUSSION THREAD

**Discussion by chiavladimirs** [swar/Swar-Chia-Plot-Manager#1206] [Q&A]
Hello all
Help with optimizing config.
Xeon E5-2673v3 2,4-2,7x2 (cpu cores 24 ), ram 128Gb, Temp SSD 2Gbx3 PNY, Storage HDD 12 Gbx3 WD

**Comment by chiavladimirs** [comment]
I think for each ssd here is the file configuration??? for SSD 2 Gb Who will help?
name: test
    max_plots: 8
    farmer_public_key:
    pool_public_key:
    temporary_directory:
      - F:\
    temporary2_directory:
    destination_directory:
      - G:\
    size: 32
    bitfield: true
    threads: 16
    buckets: 128
    memory_buffer: 4000
    max_concurrent: 8
    max_concurrent_with_start_early: 8
    initial_delay_minutes: 0
    stagger_minutes: 1
    max_for_phase_1: 4
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false
    exclude_final_directory: false
    skip_full_destinations: true
    windows_process_priority: 32
    enable_cpu_affinity: false
    cpu_affinity: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]

**Reply by chiavladimirs** [reply]
Somebody help me, please!

**Comment by chiavladimirs** [comment]
Somebody help me, please!

**Comment by w4d3j** [comment]
Have you read any of the other posts asking the exact same question?  There are literally thousands of them.

Here's one tip, which I learned from reading other posts asking this question:  using more than 4 threads sees diminishing returns, only use 4 threads per plot.  You have 8 concurrent plots each with 16 threads.  Only Phase 1 is multithreaded, but you have max_for_phase_1 set to 4 which means that on this one drive you are allocating 64 processor threads (32 cores), and as plots progress you would ultimately have 68 threads allocated (16 * 4 in Phase 1, 4 more in max_concurrent @ 1 thread each, total of 8 concurrent). That's PER DRIVE if you have more drives with this config.

That's just the first thing I noticed.  Learn your hardware, learn how the software utilizes it, and please ask more specific questions than "I don't know how to do it please somebody do this for me".

**Reply by chiavladimirs** [reply]
I have listed the equipment. There is no normal description of the parameters and how to set them, taking into account the number of threads, RAM and SSD capacity. According to this description of the equipment, a knowledgeable person can correctly suggest it. So as not to waste a lot of time. But it is written where you can ask such a question.
Support / Questions
Please do not use GitHub issues for questions or support regarding your own personal setups. Issues should pertain to actual bugs in the code and ideas. It has been tested to work on Windows, Linux, and Mac OS by numerous people at this point. So many questions relating to tech support, configuration setup, or things pertaining to your own personal use cases should be posted at any of the links below.

Discord Server: https://discord.gg/XyvMzeQpu2
This is the Official Discord Server - Swar's China Community
Officiel China Key base Team: https://keybase.io/team/chia_network.public
The channel is #war
GitHub Discussion Board: https://github.com/swar/Swar-Chia-Plot-Manager/discussions
If you are too lazy or don't know what to say, then you don't have to write anything. Not

**Reply by chiavladimirs** [reply]
I do not understand, from the description it turns out that this is the number of threads that are allocated to the plotter and not to create each raft. Is that so, or did I not understand something? It turns out that you need to divide 16 by 4 and in the end by 8 and not multiply. Or what?
threads - This is the number of threads that will be assigned to the plotter. Only phase 1 uses more than 1 thread.
![image](https://user-images.githubusercontent.com/86117669/122716382-d9f8d480-d272-11eb-8279-60b1c4501635.png)
https://github.com/swar/Swar-Chia-Plot-Manager/blob/main/README.md

**Reply by w4d3j** [reply]
The use of the word "plotter" is tricky there.  It is not referring to the Plot Manager as a whole, nor is it even referring to an entire Job as a whole.  It means every time a plot gets started, that is the number of threads that are allocated to the process.

**Comment by chiavladimirs** [comment]
name: test
max_plots: 8
farmer_public_key:
pool_public_key:
temporary_directory:
- F:
temporary2_directory:
destination_directory:
- G:
size: 32
bitfield: true
threads: 8
buckets: 128
memory_buffer: 4000
max_concurrent: 8
max_concurrent_with_start_early: 8
initial_delay_minutes: 0
stagger_minutes: 1
max_for_phase_1: 2
concurrency_start_early_phase: 2
concurrency_start_early_phase_delay: 0
temporary2_destination_sync: false
exclude_final_directory: false
skip_full_destinations: true
windows_process_priority: 32
enable_cpu_affinity: false
cpu_affinity: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]

**Comment by chiavladimirs** [comment]
So far, I have done this for each ssd. If someone knows how to do it better, write. Thank you very much.

name: SSD1_D
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: D:
#temporary2_directory:
destination_directory: E:\Plots
size: 32
bitfield: true
threads: 16
buckets: 128
memory_buffer: 4000
max_concurrent: 8
max_concurrent_with_start_early: 8
initial_delay_minutes: 0
stagger_minutes: 10
max_for_phase_1: 4
concurrency_start_early_phase: 2
concurrency_start_early_phase_delay: 0
temporary2_destination_sync: false
exclude_final_directory: false
skip_full_destinations: true
#unix_process_priority: 10
windows_process_priority: 32
enable_cpu_affinity: false
cpu_affinity: [ 0, 1, 2, 3, 4, 5, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47 ]
